compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -34,25 +34,6 @@ impl ItemLowerer<'_, '_, '_> {
 }
 
 impl<'a> Visitor<'a> for ItemLowerer<'a, '_, '_> {
-    fn visit_mod(&mut self, m: &'a Mod, _s: Span, _attrs: &[Attribute], n: NodeId) {
-        let def_id = self.lctx.lower_node_id(n).expect_owner();
-
-        self.lctx.modules.insert(
```

Comment:
Could this be moved to the `ItemKind::Mod(..) => ` match arm below? That would allow you to avoid using `lctx.modules.entry` in some of the other code.

Reply:
I think `entry` is preferable, why pre-inserting empty values explicitly if `entry` can do everything for you automatically.

Reply:
I thought that the `get_mut().unwrap()` might have been a check to ensure that we had already processed the module in some way. If that's not the case, then I think your change is fine.